torch/distributed/rpc/functions.py

```diff
@@ -49,6 +49,32 @@ def async_execution(fn):
         >>>     args=("worker2", torch.ones(2), 1, 1)
         >>> )
         >>> print(ret)  # prints tensor([3., 3.])
+
+        When combined with TorchScript functions, this decorator must be the
+        outmost one.
```

Comment:
nit: should we also mention that if we're using any other decorators, this one should still be the outermost one.